webapp/src/main/java/com/box/l10n/mojito/rest/commit/CommitWS.java

```diff
@@ -98,17 +100,72 @@ public Page<Commit> getCommits(@RequestParam(value = "repositoryId") Long reposi
     /**
      * Gets the last known commit that we have processed and recorded a
      * PushRun for from the list of commit names provided.
+     * <p>
      * Returns null if no commit is found for that repository with a
      * corresponding push run.
      *
-     * @param lastPushedCommitBody The commit names to search for together with the repositoryID.
+     * @param commitListWithRepositoryIdBody The commit names to search for together with the repositoryID.
      * @return {@link View.Commit}
      */
     @JsonView(View.Commit.class)
     @RequestMapping(value = "/api/commits/lastPushed/", method = RequestMethod.POST)
-    public Commit getLastPushedCommit(@RequestBody LastPushedCommitBody lastPushedCommitBody) {
-        return commitService.getLastPushedCommit(lastPushedCommitBody.getCommitNames(),
-                                                 lastPushedCommitBody.getRepositoryId())
+    public Commit getLastPushedCommit(@RequestBody CommitListWithRepositoryIdBody commitListWithRepositoryIdBody) {
+        return commitService.getLastPushedCommit(commitListWithRepositoryIdBody.getCommitNames(),
+                                                 commitListWithRepositoryIdBody.getRepositoryId())
+                .orElse(null);
```

Comment:
what is this actually going to return? Should it be remapped to an HTTP error?

Reply:
This should not return an error (404) as it's expected that no valid commit exists for the list of commit names provided. (E.g.: I could pass in a list of local commit hashes that Mojito never processed a push on - that's not an error situation).
When a valid commit is not found, the client is expected to loop back and call this API with additional older commit names until something is found (or a max threshold is reached).

This is the same scenario as searching the Mojito workbench / text units search API for content that doesn't exist.